### PR TITLE
finished the final rejection feature

### PR DESCRIPTION
**Pull request by SDOsmany**
final rejecting feature with the new parameter integration

### CODE REVIEW

File: base_eeg/test.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    epochs.set_montage(easy_montage)
+
+    #Sample user parameter 
+    user_params={'tmin':-0.2,'tmax':0.5}
+    output_dict_finalRej=fr.final_reject_epoch(user_params, epochs, raw)
+    print('this is the the output dictionary ---> ',output_dict_finalRej)
+
+if __name__=='__main__':
+    main()
+
```
Comment (F-said) [inline base_eeg/test.py:26]: final_reject works great for matching_pennies, nice work on annotations
```
154 bad epochs dropped
defaultdict(<class 'dict'>, {
'interpolatedChannels': 
    ['C3', 'CP5', 'CP1', 'C4', 'CP2', 'CP6'], 
'epochsDropped': 
    {'FC5': '0', 'FC1': '0', 'C3': '11', 'CP5': '79', 'CP1': '33', 'FC2': '0', 'FC6': '0', 'C4': '32', 'CP2': '54', 'CP6': '154'}
})
```

Final_reject fails for new BIDS data from feat-io. New BIDS data might need to be looked at closer:

```
ValueError: No matching events found for 82 (event id 10004)
```

**Comment by DMRoberts** [comment]
@georgebuzzell I'll check it out.


@F-said I think the error is related to the problem of the final event being outside the range of the data (the potential MNE bug we discovered last week). I tried the script with the existing dataset on the NDC Google Drive (specifically the 'ContrastChangeBlock1' task) and received the same error. However the error is resolved with the updated data from the updated BIDS import script, which eliminated those final events.

Here is the updated BIDS.zip on Google Drive if to reflect the data produced by the updated BIDS import script, if you'd like to try it again: https://drive.google.com/file/d/1DJNbJysakixs2Jr-y4GXTA2UW7e-Dte0/view?usp=sharing

That is just in my own Google Drive, I don't have the permissions at the moment to update BIDS.zip in the NDC Drive folder.

You probably already figured this out but for our BIDS example data you'll need to change the montage label to:

`easy_montage = mne.channels.make_standard_montage('GSN-HydroCel-129')
`

@SDOsmany One thing I noticed is that I think the `autoreject` package may need to be added to the Anaconda environment .yml here.

**Comment by SDOsmany** [comment]
@DMRoberts Thanks a lot for looking into this. You are right , I need to add the autoreject package. 
I will do that along with the things i need to fix

**Comment by F-said** [comment]
@DMRoberts Thanks! Made a PR to update the BIDS dataset in the readme @georgebuzzell.

**Comment by georgebuzzell** [comment]
Thank you @DMRoberts !

**Pull request closed without merging**